feat: Tesseract can support several languages

gromanas opened the pull request:
update the necessary files to support tesseract multilanguages

Contributes to: wswebcreation#16

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas committed:
feat: tesseract language option

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas committed:
feat: tesseract language option

- Update unit tests

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas committed:
feat: tesseract language option

- Update unit test snapshot

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas committed:
feat: tesseract language option

- Update unit tests

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas committed:
feat: tesseract language option

- Update unit tests

Signed-off-by: Georgios Romanas gromanas@gmail.com

wswebcreation reviewed (changes requested):
Hi @gromanas 

Wow, thanks for the PR, great work. I have some comments, but the main feedback is that this means that we have a breaking change. People now always need to add the language. My suggestion would be to make it optional for all commands:

```ts
tesseractLang?: string;
```

Make English the default

```ts
// src/service.ts
tesseractLang: this._ocrLanguage || 'eng',
```

and type it so people know which languages are supported. See an example [here](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-types/src/Options.ts#L216)

When you use the type you can also default `eng` to the type.

gromanas committed:
feat: tesseract language option

- Fix as per comments

Contributes to: #16

Signed-off-by: Georgios Romanas gromanas@gmail.com

wswebcreation commented on website/docs/getting-started.md:
```diff
@@ -66,6 +67,7 @@ The following configuration options are supported and are all optional.
 | Option | Default | Description |
 | --- | --- | --- |
 | ocrImagesPath | `{project-root}/.tmp` | The folder where the OCR-results are stored |
+| ocrLanguage | `eng` | The language that Tesseract will recognize
```
Maybe link to the location where the languages are so people know what they can use

gromanas replied:
👍🏻

wswebcreation commented on website/docs/what-is-wdio-ocr-service.md:
```diff
@@ -27,7 +27,7 @@ This service will
 1. uses [Optical Character Recognition](https://en.wikipedia.org/wiki/Optical_character_recognition) from
    [Tesseract](https://github.com/tesseract-ocr/tesseract) to get all text from the screen and highlight all found text
-   on an image
+   on an image. Can support several languages, all supported languages can be found [here.](https://tesseract-ocr.github.io/tessdoc/Data-Files-in-different-versions.html)
```
👍

wswebcreation reviewed (changes requested):
Hi @gromanas 

Again, thanks for the adjustments, I have a small remark and one test I would love to see added. It would be great if every command has a test where the language option is provided (not english) and one where it isn't. Then that statement is covered everywhere

gromanas committed:
feat: tesseract language option

-  Fix comments
-  Add new unit tests

Contributes to: #16

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas committed:
feat: tesseract language option

-  Fix comments
-  Add new unit tests

Contributes to: #16

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas committed:
feat: tesseract language option

-  Add new unit tests

Contributes to: #16

Signed-off-by: Georgios Romanas gromanas@gmail.com

wswebcreation commented on src/commands/ocrClickOnText.ts:
```diff
@@ -5,6 +5,7 @@ import { Rectangles, ScreenSize } from '../typings/types'
 interface OcrClickOnTextOptions {
   androidRectangles?:Rectangles;
   iOSRectangles?:Rectangles;
+  tesseractLang?: string;
```
Hi @gromanas 

I've been thinking about this again, and I made a mistake. All command-file must have this as a mandatory option, so `tesseractLang?: string;` needs to be `tesseractLang: string;`. This also means that all tests that you've added (for the optional ones) are not needed and need to be removed from this PR.

The tests need to be added to every command in the `service.test.ts`-file.

Again, sorry for not seeing this the first time and causing you extra work (but hopefully you learned from it 😆 )

gromanas committed:
feat: tesseract language option

- Update unit tests
- Refactor ocr language

Contributes to: #16

Signed-off-by: Georgios Romanas gromanas@gmail.com

wswebcreation reviewed (changes requested):
Hi @gromanas 

This looks really good, can you only add service tests for providing and not providing the language?

gromanas commented:
> Hi @gromanas
> 
> This looks really good, can you only add service tests for providing and not providing the language?

Hello @wswebcreation ! Thank you for the feedback and the comments. I am on it!

gromanas committed:
feat: tesseract language option

- Update unit tests and snapshots

Contributes to: #16

Signed-off-by: Georgios Romanas gromanas@gmail.com

gromanas commented:
> Hi @gromanas
> 
> This looks really good, can you only add service tests for providing and not providing the language?

I think

gromanas commented:
> > Hi @gromanas
> > This looks really good, can you only add service tests for providing and not providing the language?
> 
> Hello @wswebcreation ! Thank you for the feedback and the comments. I am on it!

I think I have added the tests to include also language. I have added also in the same tests a test to check a different ocr images path.

wswebcreation commented:
Thanks man for all your hard work on this!
